lib/util.js

```diff
@@ -373,20 +373,37 @@ function _fetch_rss_json (feed_url) {
     });
 }
 
+function _is_wordpress_feed(feed_json) {
+    const has_generator = feed_json.meta.generator &&
+          feed_json.meta.generator.startsWith('https://wordpress.org/');
+    const has_xmlns = feed_json.meta['rss:site'] &&
+          feed_json.meta['rss:site']['@'] &&
+          feed_json.meta['rss:site']['@']['xmlns'] &&
+          feed_json.meta['rss:site']['@']['xmlns'].startsWith('com-wordpress');
+    return has_generator || has_xmlns;
+}
+
 function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date) {
-    let feed_obj, is_paginated;
-    if (typeof feed === 'function') {
-        is_paginated = true;
-        feed_obj = feed(page);
+    let is_array = false;
+    let is_paginated = false;
+    let feed_url, feed_obj;
+
+    if (Array.isArray(feed)) {
+        is_array = true;
+        feed_obj = feed[0];
     } else {
-        is_paginated = false;
         feed_obj = feed;
     }
 
-    let feed_url = feed_obj, is_array = false;
-    if (Array.isArray(feed_obj)) {
-        is_array = true;
-        feed_url = feed_obj.shift();
+    if (typeof feed_obj === 'function') {
+        is_paginated = true;
+        feed_url = feed_obj(page);
+    } else {
+        feed_url = feed_obj;
+    }
```

Comment:
You could remove the `else` if you set feed_url to feed_obj before the if (reduces nesting)